src/PackageSelection/PackageSelection.php

```diff
@@ -461,4 +491,26 @@ private function filterRepositories(array $repositories)
             return true;
         });
     }
+
+    /**
+     * @param Pool $pool
+     * @param $name
+     * @param Link $link
+     * @return array|\Composer\Package\PackageInterface[]
+     */
+    private function findMatches(Pool $pool, $name, Link $link)
+    {
+        try {
+            if (!$this->bestCandidateStrategy) {
+                return $pool->whatProvides($name, $link->getConstraint(), true);
+            } else {
+                $selector = new VersionSelector($pool);
+                if ($best = $selector->findBestCandidate($name, str_replace(['[', ']'], ['', ''], $link->getConstraint()->getPrettyString()))) {
```

Comment:
That is why I propose this solution here, I couldn't find other way to have the right version, so I had to clean it. And right now it's working